Release/3.2

shilman opened the pull request:
Issue: N/A

## What I did

Placeholder for 3.2 release

## How to test

TBD

ndelangen committed:
WIP

ndelangen committed:
WIP, webpack succesful

ndelangen committed:
testing

alexandrebodin committed:
Implement basic working version, Cannot handle extension and global custom components

kazupon committed:
:shirt: refactor(client): update preview rendering

kazupon committed:
:shirt: refactor(examples): update stories

ndelangen committed:
Publish alpha.0

alexandrebodin committed:
Make global examples work if config.js bootstrap components and extensions

alexandrebodin committed:
Make working examples for functions, Vue extensions and Vue global components

alexandrebodin committed:
fix dep for lerna bootstrap

ndelangen committed:
HMR works!

ndelangen committed:
CHANGE favicon

ndelangen committed:
ADD example with actions and links

alexandrebodin committed:
Update the dep value to work with next releases of @storybook/vue

ndelangen committed:
Merge branch 'master' into add-app-vue

alexandrebodin committed:
Add support for vue in addon-notes

alexandrebodin committed:
Make addon notes agnostic

alexandrebodin committed:
deprecate old WithNotes but keep backward compatible

alexandrebodin committed:
Cleaning up deps and comments

ndelangen committed:
Merge pull request #1278 from storybooks/addon-notes-vue

Add support for vue in addon-notes

alexandrebodin committed:
Make a working example

alexandrebodin committed:
Cleaning up

kazupon committed:
:construction: wip: add story template for vue app project

alexandrebodin committed:
less ugly working example

alexandrebodin committed:
Comment unused to make PR clear

alexandrebodin committed:
Add support for addonKnobs, rename withX to addonX to avoid name collision

kazupon committed:
:up: update(cli): getstorybook vue supporting

ndelangen committed:
Merge branch 'master' into add-app-vue

# Conflicts:
#	examples/cra-kitchen-sink/.storybook/addons.js
#	examples/cra-kitchen-sink/package.json

kazupon committed:
:up: update(cli): stories creating for vue

kazupon committed:
Merge branch 'add-app-vue' into getstorybook-support-for-vue

ndelangen committed:
Change docs to contains Vue info

ndelangen committed:
Merge branch 'master' into add-app-vue

alexandrebodin committed:
Finish cleanup of addon knobs

alexandrebodin committed:
More clean up

kazupon committed:
:shirt: refactor: tweak style guide

NOTE:
  Vue.js standard style

alexandrebodin committed:
Fix example using a function as a component to object component

alexandrebodin committed:
Make example a proper vue app

alexandrebodin committed:
Readme fix

alexandrebodin committed:
Wording

kazupon committed:
:up: update: show the error at storybook/vue

kazupon committed:
:pencil: docs: update vue guide

alexandrebodin committed:
Merge pull request #1287 from kazupon/getstorybook-support-for-vue

CLI support for Vue

alexandrebodin committed:
Remove unwatend packe-lock.json

alexandrebodin committed:
Rename vue example to vue-kitchen-sink

alexandrebodin committed:
Merge branch 'master' into add-app-vue

kazupon committed:
:up: update: fix error display

alexandrebodin committed:
add more examples with knobs fields

alexandrebodin committed:
Merge branch 'add-app-vue' of github.com:storybooks/storybook into add-app-vue

ndelangen committed:
Merge branch 'master' into add-app-vue

alexandrebodin committed:
Merge branch 'master' into add-app-vue

alexandrebodin committed:
add tests

alexandrebodin committed:
fix dependency

alexandrebodin committed:
bumping version

alexandrebodin committed:
bumping version

alexandrebodin committed:
Merge branch 'add-app-vue' of github.com:storybooks/storybook into add-app-vue

alexandrebodin committed:
fix resolve in webpack

matt-oconnell committed:
Update global hook for Vue Devtools

Commit:
Add left panel hierarchy support

Commit:
Fix a few related codebeat issues

Commit:
Add more tests for stories.js

Commit:
Better UI for the hierarchy view + bugs fixes

Commit:
Merge remote-tracking branch 'origin/master' into left-panel-hierarchy

Conflicts:
	lib/ui/package.json

joscha committed:
remove typings for @storybook/addon-knobs, references https://github.com/DefinitelyTyped/DefinitelyTyped/issues/17399

joscha committed:
remove typings for @storybook/addon-links, references https://github.com/DefinitelyTyped/DefinitelyTyped/issues/17406

joscha committed:
remove typings for @storybook/addon-options, references https://github.com/DefinitelyTyped/DefinitelyTyped/pull/17408

joscha committed:
remove typings for @storybook/addon-notes, references https://github.com/DefinitelyTyped/DefinitelyTyped/pull/17409

joscha committed:
Merge branch 'master' into joscha/remove-addon-knobs-types

Commit:
Add shortcut to toggle hierarchy

igor-dv committed:
Extract hierarchy creation to the mapper

ndelangen committed:
Merge branch 'master' into add-app-vue

# Conflicts:
#	.gitignore
#	addons/knobs/package.json
#	addons/notes/package.json
#	lib/cli/package.json

shilman committed:
Merge branch 'master' into release/3.2

ndelangen committed:
WIP

alexandrebodin committed:
Merge branch 'master' of github.com:storybooks/storybook into add-app-vue

alexandrebodin committed:
clean up build and wording

alexandrebodin committed:
bump to alpha 4

igor-dv committed:
Remove shortcuts for the hierarchy toggling + change leftPanelHierarchy to resolveStoryHierarchyNamespace

igor-dv committed:
Rename resolveStoryHierarchyNamespace to resolveStoryHierarchy

ndelangen committed:
FIX multiple version react in vue app

ndelangen committed:
CLEANUP

igor-dv committed:
Fix tests + Add tests for hierarchy

ndelangen committed:
FIX multiple versions of react & vue && CLEANUP

shilman committed:
Merge pull request #1342 from joscha/joscha/remove-addon-links-types

remove typings for @storybook/addon-links

shilman committed:
Merge pull request #1339 from joscha/joscha/remove-addon-knobs-types

remove typings for @storybook/addon-knobs

shilman committed:
Merge pull request #1343 from joscha/joscha/remove-addon-options-types

remove typings for @storybook/addon-options

shilman committed:
Merge pull request #1344 from joscha/joscha/remove-addon-notes-types

remove typings for @storybook/addon-notes

igor-dv committed:
Fix the cases when the separator is not a "."

igor-dv committed:
Add assertions to left_panel.test.js

igor-dv committed:
Fix warnings in stories.test.js

igor-dv committed:
Fix tests + Add tests

igor-dv committed:
Remove require.resolve from libs/ui webpack.config.js

igor-dv committed:
Remove package-lock.json files

shilman committed:
Merge pull request #1329 from igor-dv/left-panel-hierarchy

Add left panel hierarchy support

igor-dv committed:
Try to use react-treebeard

igor-dv committed:
Try to save local state of the nodes inside the stories.js

igor-dv committed:
Alow animations (with warnings) + Styling

igor-dv committed:
Repalce resolveStoryHierarchy func with hierarchySeparator

igor-dv committed:
Merge remote-tracking branch 'origin/151-story-hierarchy' into story-hierarchy-3rd-party

Merge remote-tracking branch 'origin/151-story-hierarchy' into story-hierarchy-3rd-party

igor-dv committed:
Delete unused code + adjust names

igor-dv committed:
Fix hierarchy + remove unused functionality

kazupon committed:
:up: update(server): tweak consoleoutputing

igor-dv committed:
Fix stories.test.js

igor-dv committed:
Remove the root node of the tree

igor-dv committed:
Add react-icons for a better tree structure

igor-dv committed:
Change the order of the atomic namespace

shilman committed:
Merge branch 'master' into release/3.2

igor-dv committed:
Merge remote-tracking branch 'origin/release/3.2' into 151-story-hierarchy

Conflicts:
	addons/info/README.md

igor-dv committed:
Revert code that was injected by aliens

igor-dv committed:
Merge remote-tracking branch 'origin/151-story-hierarchy' into story-hierarchy-3rd-party

igor-dv committed:
Merge pull request #1356 from storybooks/151-story-hierarchy

Story Hierarchy (continued)

igor-dv committed:
Remove a redundant comment

igor-dv committed:
Move decorators to separate file

igor-dv committed:
Fix a bug when the component node and one of the namespaces nodes have the same name

igor-dv committed:
Extract tree_style to separate file

alexandrebodin committed:
Merge branch 'add-app-vue' of github.com:storybooks/storybook into add-app-vue

alexandrebodin committed:
Merge branch 'master' of github.com:storybooks/storybook into add-app-vue

alexandrebodin committed:
fix version in package.json

igor-dv committed:
Use a forked version of the react-treebeard

shilman committed:
Merge branch 'master' into release/3.2

shilman committed:
Merge branch 'release/3.2' into 151-story-hierarchy

shilman committed:
Merge remote-tracking branch 'origin/151-story-hierarchy' into 151-story-hierarchy

shilman committed:
Merge branch 'release/3.2' into add-app-vue

igor-dv committed:
Rename stories dir to the stories_tree to prevent prepublish script ignore flags collision

shilman committed:
Merge pull request #1267 from storybooks/add-app-vue

WIP - add vue support

igor-dv committed:
Upgrade storybook-react-treebeard to a newer version that fixes import syntax

igor-dv committed:
Add more tests for stories_tree

shilman committed:
Merge pull request #1387 from storybooks/151-story-hierarchy

Story Hierarchy (UI improvements)

shilman committed:
v3.2.0-alpha.5

shilman committed:
Update cra-kitchen-sink versioning for alpha release

`-alpha` naming scheme is not semver, so Learn does not do the right
thing here, and as a workaround we need to hardcode the version numbers
while we’re doing alpha releases if we want our example apps to work
properly.

shilman committed:
Fix typo

igor-dv committed:
Fix loss of the state in a reall app with componentWillReceiveProps

shilman committed:
Merge pull request #1400 from storybooks/examples-3.2-alpha-versioning

Update cra-kitchen-sink versioning for alpha release

shilman committed:
Merge branch 'release/3.2' into 151-story-hierarchy

igor-dv committed:
Add tests to check the componentWillReceiveProps

igor-dv committed:
Merge remote-tracking branch 'origin/release/3.2' into 151-story-hierarchy

alexandrebodin committed:
Bump vue to 3.2.0-alpha.6 to publish as latest (getstorybook needs this)

igor-dv committed:
Use deepEqual to compare selectedHierarchy

igor-dv committed:
Merge remote-tracking branch 'origin/release/3.2' into 151-story-hierarchy

matt-oakes committed:
MVP version of the on device React Native UI

This is a combination of 10 commits.

- Add a basic implementation of the native UI for the storybook
- Fix the lint errors
- Add a param to turn the native UI on and off using the same package
- Correctly set the width of the story list
- Added a size selector for the different sized iPhone screens
- Show the selected story in bold
- Add some styling to the panels
- Removed the canvas size switcher
- Fix lint errors
- Shortened the long render item and header lines

shilman committed:
Merge remote-tracking branch 'origin/151-story-hierarchy' into 151-story-hierarchy

shilman committed:
Merge pull request #1401 from storybooks/151-story-hierarchy

Story Hierarchy - initial state bug fix

shilman committed:
Merge pull request #1419 from storybooks/master

Updating 3.2 alpha release with patches

ndelangen committed:
Merge branch 'release/3.2' into update-global-hook

shilman committed:
v3.2.0-alpha.7

shilman committed:
v3.2.0-alpha.7

igor-dv committed:
Add keyboard accessibility to stories tree

igor-dv committed:
Merge branch '151-story-hierarchy' of https://github.com/storybooks/storybook into 151-story-hierarchy

igor-dv committed:
Merge remote-tracking branch 'origin/release/3.2' into 151-story-hierarchy

igor-dv committed:
Fix long story name word-breaking

igor-dv committed:
Add test for pressing enter on story

jackhurley23 committed:
Implement filtering on story-level

jackhurley23 committed:
Update filtering to be case-insensitive at story level

shilman committed:
Update cra-kitchen-sink package versions for 3.2-alpha

ndelangen committed:
Merge pull request #1434 from storybooks/1433-fix-alpha-versions

Update cra-kitchen-sink package versions for 3.2-alpha

usulpro committed:
Refactoring Addon Info [WIP]

usulpro committed:
Addons composition

igor-dv committed:
Merge remote-tracking branch 'origin/release/3.2' into 151-story-hierarchy

joscha committed:
Add hierarchySeparator to README

usulpro committed:
Add withInfo()(storyFn)

usulpro committed:
Fix addon notes

igor-dv committed:
Fix the left panel width when narrowing

usulpro committed:
Merge pull request #1448 from storybooks/Fix-addon-notes

Fix addon notes

ndelangen committed:
Merge branch 'release/3.2' into joscha/add-hierarchySeparator

igor-dv committed:
Merge remote-tracking branch 'origin/release/3.2' into 151-story-hierarchy

usulpro committed:
Merge branch 'release/3.2' into 1147-info-addon

usulpro committed:
Add deprecation warning

usulpro committed:
Refactoring withInfo function

usulpro committed:
Add test

usulpro committed:
Improve test coverage

usulpro committed:
Fix typo

usulpro committed:
Switch to util-deprecate

igor-dv committed:
Mark the nested prop types with isRequired

usulpro committed:
Fix tests and default props

shilman committed:
Merge pull request #1452 from storybooks/1147-info-addon

Refactoring of `addon-info`

shilman committed:
Merge branch 'release/3.2' into 151-story-hierarchy

igor-dv committed:
Merge pull request #1427 from storybooks/151-story-hierarchy

Story Hierarchy - keyboard accessibility

ndelangen committed:
Merge branch 'release/3.2' into joscha/add-hierarchySeparator

joscha committed:
cahnge to /

shilman committed:
Merge branch 'master' into release/3.2

# Conflicts:
#	addons/actions/package.json
#	addons/knobs/package.json
#	addons/storyshots/package.json
#	app/react-native/package.json
#	app/react/package.json
#	examples/cra-kitchen-sink/package.json
#	lerna.json
#	lib/ui/src/modules/ui/components/layout/index.js

shilman committed:
Merge pull request #1445 from storybooks/joscha/add-hierarchySeparator

Add hierarchySeparator to README

shilman committed:
Merge pull request #1432 from jhurley23/109-story-level-filtering

Implement filtering on story-level

kazupon committed:
:pushpin: dependency(vue): bump vue

kazupon committed:
:shirt: refactor(vue): tweak vue implementation

kazupon committed:
:pushpin: dependency(examples): bump dependency modules

ndelangen committed:
Merge branch 'release/3.2' into update-global-hook

igor-dv committed:
Set '/' as a default hierarchySeparator

igor-dv committed:
Remove '/' from ui\example\client\provider since it's a default value

igor-dv committed:
Change the react-treebeard to the main repo version

ndelangen committed:
Merge branch 'release/3.2' into storybook-native-ui

igor-dv committed:
Change toggle icon and delete other icons to make the tree look cleaner :cry:

shilman committed:
Merge pull request #1413 from matt-oakes/storybook-native-ui

React Native - On Device UI

UsulPro commented:
Is it a known issue with HMR?

How to reproduce:

- run `cra-kitchen-sink`

- select any story in Storybook

- edit something in `src/stories/index.js`

- save file and check out how HMR updates the page.

Expected (@3.1.x behavior):

Storybook should keep selected story opened and the change should be applied

Current (this branch) behavior:

Storybook jumps to the first story but Stories Panel still shows the previous story selected.

ndelangen committed:
Merge branch 'release/3.2' into 151-story-hierarchy

ndelangen committed:
Merge branch 'release/3.2' into tweak/vue-app

igor-dv committed:
Merge pull request #1488 from storybooks/151-story-hierarchy

Trying to consolidate hierarchy UI (+ some other fixes)

ndelangen committed:
Merge branch 'master' into release/3.2

# Conflicts:
#	addons/actions/package.json
#	addons/info/package.json
#	addons/knobs/package.json
#	addons/storyshots/package.json
#	app/react-native/package.json
#	app/react/package.json
#	lerna.json
#	lib/cli/package.json
#	lib/ui/package.json

ndelangen committed:
linting

ndelangen committed:
UPDATE / SYNC package dependencies for optimal bootstrap

ndelangen committed:
FIX sometimes storyName being undefined when url is '/'

joscha commented on addons/info/src/index.js:
```diff
@@ -29,59 +34,62 @@ const defaultMarksyConf = {
-        maxPropsIntoLine: options.maxPropsIntoLine,
-        maxPropStringLength: options.maxPropStringLength,
-      };
+export const withInfo = (info, _options) =>
```
- [ ] This will need a followup in https://github.com/DefinitelyTyped/DefinitelyTyped/

joscha commented on addons/knobs/package.json:
```diff
@@ -37,7 +37,8 @@
     "style-loader": "^0.17.0",
     "typescript": "^2.2.2",
-    "typescript-definition-tester": "^0.0.5"
+    "typescript-definition-tester": "^0.0.5",
```
do we still need this, now that we have moved the typescript definitions?

- [ ] @joscha to follow up

joscha commented on addons/knobs/src/vue/index.js:
```diff
@@ -0,0 +1,37 @@
+    channel.removeListener('addon:knobs:knobChange', this.onKnobChange);
+    knobStore.unsubscribe(this.setPaneKnobs);
+  }
+});
```
nit: newline missing?

joscha commented on addons/options/README.md:
```diff
@@ -44,6 +44,7 @@ setOptions({
   showSearchBox: false,
   downPanelInRight: false,
   sortStoriesByKind: false,
+  hierarchySeparator: /\//,
```
- [ ] needs follow-up on https://github.com/DefinitelyTyped/DefinitelyTyped/

joscha replied:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/18253

igor-dv replied:
@joscha , currently `hierarchySeparator` is a string value, that represents regex, but not the RegExp object (because when serializing RegExp it becomes empty object, and everything in `setOptions` should be serializable) 

@UsulPro has a nice solution for that here -
 https://github.com/storybooks/storybook/pull/1394/files#r126082393

joscha replied:
ah cool, this would get applied automatically - what are the chances this PR will make it into 3.2? if the chances are low I will adapt the readme and the definitelytyped PR.

igor-dv replied:
I will try to get it in.

joscha commented on addons/notes/src/index.js:
```diff
@@ -1,24 +1,22 @@
+  enumerable: true,
+  get: deprecate(
+    () => ReactWithNotes,
+    '@storybook/addon-notes WithNotes Component is deprecated, use withNotes() instead. See https://github.com/storybooks/storybook/tree/master/addons/notes'
```
- [ ] needs follow-up on https://github.com/DefinitelyTyped/DefinitelyTyped/

joscha reviewed (approved):
Some follow-ups needed for the Typescript types, nothing major however, so from that perspective good for a release. Skipped everything vue and treebeard.

ndelangen committed:
Merge branch 'release/3.2' into tweak/vue-app

kazupon committed:
:pushpin: dependecy:make vue-template-compiler the same as vue core

shilman committed:
Merge pull request #1477 from kazupon/tweak/vue-app

Vue: Tweak vue app

shilman committed:
v3.2.0-alpha.8

shilman committed:
Fix vue release and examples dependencies

ndelangen committed:
Merge branch 'master' into release/3.2

shilman commented:
@joscha none of this stuff is final -- will probably have a few more changes before we release, so please don't pull the trigger on any of this until we are all synced.

ndelangen committed:
Extract nopreview component & allow it to be re-rendered

igor-dv committed:
Fix a 'funny' hmr issue

ndelangen committed:
FIX unittests

ndelangen committed:
Merge branch 'release/3.2' into fix-nopreview-vue

ndelangen committed:
Merge branch 'release/3.2' into fix-hmr-issue-in-cra-kitchen-sink

ndelangen committed:
Merge pull request #1504 from storybooks/fix-nopreview-vue

Fix nopreview vue

ndelangen committed:
Merge branch 'release/3.2' into fix-hmr-issue-in-cra-kitchen-sink

ndelangen committed:
Merge pull request #1508 from storybooks/fix-hmr-issue-in-cra-kitchen-sink

Fix a 'funny' hmr issue in cra-kitchen-sink

shilman committed:
Fix RN preview welcome instructions

shilman committed:
Reset storybook on startup?

@matt-oakes I’m not sure why I need to do this but I get a “URL
required” prop-types error unless I force `resetStorybook` in
`react-native-vanilla` example

shilman committed:
Show first story on OnDeviceUI startup

shilman committed:
Merge branch 'release/3.2' into 1492-on-device-startup

ndelangen committed:
Merge branch 'release/3.2' into update-global-hook

ndelangen committed:
Merge pull request #1376 from matt-oconnell/update-global-hook

Update global hook for Vue Devtools

ndelangen committed:
Update README.md

shilman committed:
Merge pull request #1510 from storybooks/1492-on-device-startup

Show first story on OnDeviceUI startup

shilman committed:
v3.2.0-alpha.9

ndelangen committed:
Merge branch 'master' into release/3.2

ndelangen committed:
Merge branch 'master' into release/3.2

ndelangen committed:
Merge branch 'master' into release/3.2

ndelangen committed:
Merge branch 'master' into release/3.2

igor-dv committed:
Add warnings if 'module' is missing

igor-dv committed:
Merge pull request #1525 from storybooks/hmr-warning-on-storiesOf

Add warning when module is missing in storiesOf

danielduan reviewed (approved):
The published alpha works well for me!

ndelangen committed:
Merge branch 'master' into release/3.2

tmeasday committed:
Revert knobs API to previous API.

`withKnobs` the decorator is still the base API. (This now works for
Vue).

The new API is called `wrapperKnobs` and will eventually be merged into
`withKnob` (c.f. medium term addons plan, to be posted)

tmeasday committed:
Remove wrapperKnobs

tmeasday committed:
Revert addonNotes -> withNotes

shilman committed:
Merge pull request #1527 from storybooks/simplifyKnobsAPI

Revert knobs API to previous API.

shilman committed:
Fix bad merge that removed react-treebeard dep from lib/ui

ndelangen committed:
SYNC dependencies for improved hoisting

shilman committed:
Merge pull request #1531 from storybooks/fix-3.2-package-versions

Fix bad merge that removed react-treebeard dep from lib/ui

shilman committed:
v3.2.0-alpha.10

igor-dv committed:
Add negative margin to move the tree to the left =/

shilman committed:
Merge pull request #1533 from storybooks/1532-fix-stories-decorator-readonly-prop-assign

Quick-fix for story hierarchy UI and react16

shilman committed:
v3.2.0-alpha.11

shilman committed:
Update example package deps

shilman committed:
v3.2.0 changelog
